src/polymer/polymer2-element-scanner.ts

```diff
@@ -462,20 +475,48 @@ class ElementVisitor implements Visitor {
       const className = astValue.getIdentifierName(expression.object);
       if (isPropertyNameIs && className) {
         return {
-          type: 'is',
-          className,
-          classNameSourceRange:
-              this._document.sourceRangeForNode(expression.object)!
+          kind: 'success',
+          value: {
+            type: 'is',
+            className,
+            classNameSourceRange:
+                this._document.sourceRangeForNode(expression.object)!
+          }
         };
       }
     }
-    this._warnings.push({
-      code: 'cant-determine-element-tagname',
-      message:
-          `Unable to evaluate this expression down to a definitive string tagname.`,
-      severity: Severity.WARNING,
-      sourceRange: this._document.sourceRangeForNode(expression)!
-    });
-    return;
+    return {
+      kind: 'failure',
+      warning: {
+        code: 'cant-determine-element-tagname',
+        message:
+            `Unable to evaluate this expression down to a definitive string tagname.`,
+        severity: Severity.WARNING,
+        sourceRange: this._document.sourceRangeForNode(expression)!
+      }
+    };
   }
 }
+
+type OrWarning<V> = {
```

Comment:
Well it'd never be an OrWarning, it'd be a `OrWarning<SpecificKindOfValue>` but it is a bit backwards reading in english.

Haskell would call this type `Either<V, Warning>`, Rust calls it `Result<V, Warning>`. I guess having `Warning` in the type name isn't any shorter really than having it as a type parameter. How about `Result`?

I also changed the name of the property on `failure` from `warning` to `value`. This will make v8 happier, as there's only one shape to our return wrapper.